Client/Frontend/Notifications/NotificationRootViewController.swift

```diff
@@ -159,9 +161,34 @@ private extension NotificationRootViewController {
         }
     }
 
-    @objc func didFinishSyncing() {
-        showingNotification = false
-        dispatch_async(dispatch_get_main_queue()) {
+    @objc func didFinishSyncing(notification: NSNotification) {
+        let syncMessage: NSAttributedString?
+        if let syncManager = notification.object as? SyncManager,
+            let syncDisplayState = syncManager.syncDisplayState {
+            switch syncDisplayState {
+            case .Bad(let message):
+                syncMessage = NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowErrorTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+            case .Stale(let message):
+                syncMessage = NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowWarningTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+            default: syncMessage = nil
+            }
+        } else {
+            syncMessage = nil
+        }
+
+        guard let displayMessage = syncMessage else {
+            showingNotification = false
+            return dispatch_async(dispatch_get_main_queue()) {
+                self.hideStatusNotification()
+            }
+        }
+        
+        notificationView.titleLabel.attributedText = displayMessage
+        notificationView.ellipsisLabel.hidden = true
+        self.notificationView.endAnimation()
+        let delayTime = dispatch_time(DISPATCH_TIME_NOW, Int64(4 * Double(NSEC_PER_SEC)))
+        dispatch_after(delayTime, dispatch_get_main_queue()) {
+            self.showingNotification = false
```

Comment:
> however with you patch it feels like I can no longer guarantee that a message will already be displayed when the sync result comes in, so I will have to rethink this anyway.

My original assumption was the same in that the notification would always be appearing while syncing but it's no longer the case. We only show the 'Syncing...' for 4 seconds and only show it if the user has manually initiated a sync. The case where we update the notification with a new title instead of show would be more of an edge case than showing a new one.